packages/components/src/internal/components/forms/input/useDataChangeCommentsRequired.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+import { useServerContext } from '../../base/ServerContext';
+import { isProductProjectsEnabled, resolveModuleContext } from '../../../app/utils';
+import { useAppContext } from '../../../AppContext';
+
+export const useDataChangeCommentsRequired = (): { requiresUserComment: boolean } => {
+    const { container, moduleContext } = useServerContext();
+    const { api } = useAppContext();
+    const [requiresUserComment, setRequiresUserComment] = useState<boolean>();
+
+    useEffect(
+        () => {
+            (async () => {
+                let path = container?.path;
+                if (isProductProjectsEnabled(resolveModuleContext(moduleContext)) && !container?.isProject) {
```

Comment:
No need to call `resolveModuleContext` here as `isProductProjectsEnabled` already does this. `moduleContext` can be passed directly.

Reply:
Oh right. Not sure what I was thinking...